### PR TITLE
Remove Rails deprecation warning

### PR DESCRIPTION
**Pull request by abrisse**

### CODE REVIEW

File: lib/request_store_rails/railtie.rb [hunk 1]
```diff
@@ -6,9 +6,15 @@ class Railtie < ::Rails::Railtie
...
-        RequestLocals.clear_all!
+     
+      if ActiveSupport.const_defined?(:Reloader) && ActiveSupport::Reloader.respond_to?(:to_complete)
+        ActiveSupport::Reloader.to_complete do
+          RequestLocals.clear_all!
+        end
+      elsif ActionDispatch.const_defined?(:Reloader) && ActionDispatch::Reloader.respond_to?(:to_cleanup)
+        ActionDispatch::Reloader.to_cleanup do
+          RequestLocals.clear_all!
+        end
```
Comment (ElMassimo) [inline lib/request_store_rails/railtie.rb]: Can we clean it up a little?

```ruby
clear_all = -> { RequestLocals.clear_all! }

if defined?(ActiveSupport::Reloader)
  ActiveSupport::Reloader.to_complete(&clear_all)
else
  ActionDispatch::Reloader.to_cleanup(&clear_all)
end
```

Comment (abrisse) [inline lib/request_store_rails/railtie.rb]: Sure. FYI this PR was a "cherry pick" of this commit : https://github.com/steveklabnik/request_store/commit/43298a51c004575758813eedb1b914ec5cd60c33

**Comment by abrisse** [comment]
I have updated the code as requested. FYI the initial PR was a cherry-pick of https://github.com/steveklabnik/request_store/commit/43298a51c004575758813eedb1b914ec5cd60c33 to maintain maximal compatibility.

**Comment by ElMassimo** [comment]
@abrisse Thank you, makes sense but I'd prefer if it doesn't fail silently when no `Reloader` is present.

**Comment by ElMassimo** [comment]
Just released [`1.0.2`](https://github.com/ElMassimo/request_store_rails/blob/master/CHANGELOG.md#requestlocals-102-2017-01-25) which includes this PR to remove the deprecation warning. Thanks again @abrisse!

**Pull request merged**